stdlib/trains/trains.lua

```diff
@@ -0,0 +1,192 @@
+--- Trains module
+-- <p>When this module is loaded, it registers new
+-- events in order to track trains as locomotives and
+-- carriages are moved around. For this reason, you should use
+-- this library's <a href="Event.html">Event</a> module</p>
+-- @module Trains
+
+-- Event registration is performed at the bottom of this file,
+-- once all other functions have been defined
+
+require 'stdlib/event/event'
+require 'stdlib/surface'
+require 'stdlib/table'
+
+Trains = {
+    _registry = {}
+}
+
+--- Event id for the custom event fired
+-- when a train has been renamed

```

Comment:
Can you clarify this statement a little bit? It's fairly important documentation, as without a clear understanding, the event won't be used, and its unclear what exactly 'renaming' a train means. If 'renaming' is the correct terminology, please define it in with regards to factorio.